service/internal/git/gitClient.go

```diff
@@ -51,43 +53,60 @@ func (g gitSvcImpl) Checkout(w *git.Worktree, opts *git.CheckoutOptions) error {
 	return w.Checkout(opts)
 }
 
+// Option is a function for configuring the BasicClient
+type Option func(*BasicClient)
+
+func WithProgressWriter(w io.Writer) Option {
+	return func(c *BasicClient) {
+		c.pw = w
+	}
+}
+
 // BasicClient connects to git using ssh
 type BasicClient struct {
 	auth    transport.AuthMethod
 	mu      *sync.Mutex
 	git     gitSvc
 	fs      fs.FS
 	baseDir string // base directory to run git operations from
+	pw      io.Writer
 }
 
 // NewSSHBasicClient creates a new ssh based git client
-func NewSSHBasicClient(sshPemFile string) (BasicClient, error) {
+func NewSSHBasicClient(sshPemFile string, opts ...Option) (BasicClient, error) {
 	auth, err := ssh.NewPublicKeysFromFile("git", sshPemFile, "")
 	if err != nil {
 		return BasicClient{}, err
 	}
 
-	return BasicClient{
-		auth:    auth,
-		mu:      &sync.Mutex{},
-		git:     gitSvcImpl{},
-		fs:      os.DirFS(os.TempDir()),
-		baseDir: os.TempDir(),
-	}, nil
+	return newBasicClient(auth, opts...), nil
 }
 
 // NewHTTPSBasicClient creates a new https based git client
-func NewHTTPSBasicClient(user, pass string) (BasicClient, error) {
-	return BasicClient{
-		auth: &http.BasicAuth{
-			Username: user,
-			Password: pass,
-		},
+func NewHTTPSBasicClient(user, pass string, opts ...Option) (BasicClient, error) {
+	auth := &http.BasicAuth{
+		Username: user,
+		Password: pass,
+	}
+
+	return newBasicClient(auth, opts...), nil
+}
+
+func newBasicClient(auth transport.AuthMethod, opts ...Option) BasicClient {
+	cl := BasicClient{
+		auth:    auth,
 		mu:      &sync.Mutex{},
 		git:     gitSvcImpl{},
 		fs:      os.DirFS(os.TempDir()),
 		baseDir: os.TempDir(),
-	}, nil
+		pw:      ioutil.Discard,
```

Comment:
Defaulting to discard unless overridden, which will happen currently in `service/main` when env.LogLevel is debug.